kodilla-testing/src/test/java/com/kodilla/testing/forum/statistics/StatisticsCalculatorTestSuite.java

```diff
@@ -0,0 +1,262 @@
+        Assert.assertEquals(0.0, statisticsCalculator.getAveragePostPerUser(), 0.0);
+        Assert.assertEquals(0.0, statisticsCalculator.getAverageCommentPerUser(), 0.0);
+        Assert.assertEquals(0.0, statisticsCalculator.getAverageCommentPerPost(), 0.0);
+    }
+
+    @Test
+    public void testCalculateAdvStatistics1000PostsOtherZeros() {
+        //Given
+        Statistics statisticsMock = mock(Statistics.class);
+        when(statisticsMock.usersNames()).thenReturn(generate0Users());
+        when(statisticsMock.postsCount()).thenReturn(generate1000Posts());
+        when(statisticsMock.commentsCount()).thenReturn(generate0Comments());
+        StatisticsCalculator statisticsCalculator = new StatisticsCalculator();
+        statisticsCalculator.calculateAdvStatistics(statisticsMock);
+        //When
+        statisticsCalculator.calculateAdvStatistics(statisticsMock);
+        //Then
+        Assert.assertEquals(0, statisticsCalculator.getUserCounter());
+        Assert.assertEquals(1000, statisticsCalculator.getPostCounter());
+        Assert.assertEquals(0, statisticsCalculator.getCommentCounter());
+        Assert.assertEquals(0.0, statisticsCalculator.getAveragePostPerUser(), 0.0);
+        Assert.assertEquals(0.0, statisticsCalculator.getAverageCommentPerUser(), 0.0);
+        Assert.assertEquals(0.0, statisticsCalculator.getAverageCommentPerPost(), 0.0);
+    }
+
+    @Test
+    public void testCalculateAdvStatistics500CommentsOtherZero() {
+        //Given
+        Statistics statisticsMock = mock(Statistics.class);
+        when(statisticsMock.usersNames()).thenReturn(generate0Users());
+        when(statisticsMock.postsCount()).thenReturn(generate0Posts());
+        when(statisticsMock.commentsCount()).thenReturn(generate500Comments());
+        StatisticsCalculator statisticsCalculator = new StatisticsCalculator();
+        statisticsCalculator.calculateAdvStatistics(statisticsMock);
+        //When
+        statisticsCalculator.calculateAdvStatistics(statisticsMock);
+        //Then
+        Assert.assertEquals(0, statisticsCalculator.getUserCounter());
+        Assert.assertEquals(0, statisticsCalculator.getPostCounter());
+        Assert.assertEquals(500, statisticsCalculator.getCommentCounter());
+        Assert.assertEquals(0.0, statisticsCalculator.getAveragePostPerUser(), 0.0);
+        Assert.assertEquals(0.0, statisticsCalculator.getAverageCommentPerUser(), 0.0);
+        Assert.assertEquals(0.0, statisticsCalculator.getAverageCommentPerPost(), 0.0);
+    }
+
+    @Test
+    public void testCalculateAdvStatistics0Users1000Posts500Comments() {
+        //Given
+        Statistics statisticsMock = mock(Statistics.class);
+        when(statisticsMock.usersNames()).thenReturn(generate0Users());
+        when(statisticsMock.postsCount()).thenReturn(generate1000Posts());
+        when(statisticsMock.commentsCount()).thenReturn(generate500Comments());
```

Comment:
`generate1000Posts()` i `generate500Comments()` nie generuje postów/komentarzy tylko zwraca liczbę, więc ta nazwa nie jest zbyt adekwatna, w tak prostym przypadku, gdzie zwracana jest liczba można to zrobić po prostu tak:
`when(statisticsMock.commentsCount()).thenReturn(500);`